blockchain/chain_sync/src/sync.rs

```diff
@@ -1052,20 +1056,24 @@ mod tests {
             )
             .unwrap(),
             event_sender,
+            test_receiver,
         )
     }
 
-    fn send_blocksync_response(event_sender: Sender<NetworkEvent>) {
+    fn send_blocksync_response(blocksync_message: Receiver<NetworkMessage>) {
         let rpc_response = construct_blocksync_response();
 
         task::block_on(async {
-            event_sender
-                .send(NetworkEvent::BlockSyncResponse {
-                    // TODO update this, only matching first index of requestId
-                    request_id: RequestId(1),
-                    response: rpc_response,
-                })
-                .await;
+            match blocksync_message.recv().await.unwrap() {
+                NetworkMessage::BlockSyncRequest {
+                    peer_id: _,
+                    request: _,
+                    response_channel,
+                } => {
+                    response_channel.send(rpc_response).unwrap();
+                }
+                _ => unreachable!(),
```

Comment:
I think if you use an if let here you might not even need the unreachable macro

Reply:
Yeah if you do this, you'd have to wrap in a loop and break when you send the response you want. Doesn't really matter much because it's a test tho

Reply:
Ah yeah since it is a test it's not really an issue